src/libsyntax/std_inject.rs

```diff
@@ -35,6 +35,9 @@ pub fn injected_crate_name() -> Option<&'static str> {
 }
 
 thread_local! {
+    // A `Symbol` might make more sense here, but it doesn't work, probably for
+    // reasons relating to the use of thread-local storage for the Symbol
+    // interner.
     static INJECTED_CRATE_NAME: Cell<Option<&'static str>> = Cell::new(None);
```

Comment:
This thing should be kept somewhere in session rather than in a thread local, IIRC one of @Zoxc's in-progress PRs already does that.

Reply:
I don't remember that =P

Reply:
In a session seems better, but it's out-of-scope for this PR so I will resolve this conversation :)